### PR TITLE
Add an explainer for GPULimits

### PR DESCRIPTION
**Pull request by JusSn**
This explainer will track the rationale behind each WebGPU limit. All queryable restrictions should have a corresponding reason in this doc.

Edit: [Rendered](https://github.com/JusSn/gpuweb/blob/limitsmd/design/Limits.md)

### CODE REVIEW

**Comment by JusSn** [comment]
Part of https://github.com/gpuweb/gpuweb/issues/483

File: design/Limits.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# GPULimits Explainer
+
+This document lists the citations for the "limits" in the WebGPU API that decide the minimum capabilities of a compliant WebGPU implementation.
+
+## The GPULimits Dictionary (10/29/19)
```
Comment (kainino0x) [inline design/Limits.md]: nit: use standard date format (even though this isn't ambiguous)
```suggestion
## The GPULimits Dictionary (last updated 2019-10-29)
```

Comment (kainino0x) [reply design/Limits.md]: I can't land the suggestion myself, but feel free to merge when ready.

File: design/Limits.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    unsigned long maxSamplersPerShaderStage = 16;
+    unsigned long maxStorageBuffersPerShaderStage = 4;
+    unsigned long maxStorageTexturesPerShaderStage = 4;
+    unsigned long maxUniformBuffersPerShaderStage = 12;
+};
+```
+
+Limit | API Doc | gpuweb issue/PR
+--- | --- | ---
+`maxBindGroups = 4;` | [Vulkan](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#limits-minmax) `maxBoundDescriptorSets` |
```
Comment (kvark) [inline design/Limits.md]: unfortunate that Vulkan spec is so heavy

File: design/Limits.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+`maxDynamicUniformBuffersPerPipelineLayout = 8;` | [Vulkan](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#limits-minmax) `maxDescriptorSetUniformBuffersDynamic` | [#406](https://github.com/gpuweb/gpuweb/issues/406)
```
Comment (kvark) [inline design/Limits.md]: Also, a bit strange to see *all* of the links pointing to Vulkan

File: design/Limits.md (hunk 2 above)
Comment (JusSn) [reply design/Limits.md]: I thought it would at least redirect to the specific section. :(

File: design/Limits.md (hunk 3 above)
Comment (JusSn) [reply design/Limits.md]: Makes sense considering Vulkan's lower-end Android hardware support. But the theory is that future limits won't be decided by Vulkan

File: design/Limits.md (hunk 2 above)
Comment (kainino0x) [reply design/Limits.md]: It does for me, if I don't touch it while it's loading.

How about the LunarG chunked version?
https://vulkan.lunarg.com/doc/view/latest/linux/chunked_spec/chap36.html#limits-minmax

Comment (kainino0x) [reply design/Limits.md]: https://vulkan.lunarg.com/doc/view/latest/linux/chunked_spec/chap36.html#limits-required

File: design/Limits.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    unsigned long maxSamplersPerShaderStage = 16;
+    unsigned long maxStorageBuffersPerShaderStage = 4;
+    unsigned long maxStorageTexturesPerShaderStage = 4;
+    unsigned long maxUniformBuffersPerShaderStage = 12;
+};
+```
+
+Limit | API Doc | gpuweb issue/PR
+--- | --- | ---
+`maxBindGroups = 4;` | [Vulkan](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#limits-minmax) `maxBoundDescriptorSets` |
```
Comment (JusSn) [inline design/Limits.md]:
```suggestion
`maxBindGroups = 4;` | [Vulkan](https://vulkan.lunarg.com/doc/view/latest/linux/chunked_spec/chap36.html#limits-required) `maxBoundDescriptorSets` |
```

File: design/Limits.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+`maxDynamicUniformBuffersPerPipelineLayout = 8;` | [Vulkan](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#limits-minmax) `maxDescriptorSetUniformBuffersDynamic` | [#406](https://github.com/gpuweb/gpuweb/issues/406)
```
Comment (JusSn) [inline design/Limits.md]:
```suggestion
`maxDynamicUniformBuffersPerPipelineLayout = 8;` | [Vulkan](https://vulkan.lunarg.com/doc/view/latest/linux/chunked_spec/chap36.html#limits-required) `maxDescriptorSetUniformBuffersDynamic` | [#406](https://github.com/gpuweb/gpuweb/issues/406)
```

Comment (JusSn) [inline design/Limits.md]:
```suggestion
`maxDynamicUniformBuffersPerPipelineLayout = 8;` | [Vulkan](https://vulkan.lunarg.com/doc/view/latest/linux/chunked_spec/chap36.html#limits-required) `maxDescriptorSetUniformBuffersDynamic` | [#406](https://github.com/gpuweb/gpuweb/issues/406)
```

File: design/Limits.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+`maxDynamicStorageBuffersPerPipelineLayout = 4;` | [Vulkan](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#limits-minmax) `maxDescriptorSetStorageBuffersDynamic` | [#406](https://github.com/gpuweb/gpuweb/issues/406)
```
Comment (JusSn) [inline design/Limits.md]:
```suggestion
`maxDynamicStorageBuffersPerPipelineLayout = 4;` | [Vulkan](https://vulkan.lunarg.com/doc/view/latest/linux/chunked_spec/chap36.html#limits-required) `maxDescriptorSetStorageBuffersDynamic` | [#406](https://github.com/gpuweb/gpuweb/issues/406)
```

File: design/Limits.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+`maxSampledTexturesPerShaderStage = 16;` | [Vulkan](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#limits-minmax) `maxPerStageDescriptorSampledImages` | [#409](https://github.com/gpuweb/gpuweb/issues/409)
```
Comment (JusSn) [inline design/Limits.md]:
```suggestion
`maxSampledTexturesPerShaderStage = 16;` | [Vulkan](https://vulkan.lunarg.com/doc/view/latest/linux/chunked_spec/chap36.html#limits-required) `maxPerStageDescriptorSampledImages` | [#409](https://github.com/gpuweb/gpuweb/issues/409)
```

File: design/Limits.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+`maxSamplersPerShaderStage = 16;` | [Vulkan](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#limits-minmax) `maxPerStageDescriptorSamplers` | [#409](https://github.com/gpuweb/gpuweb/issues/409)
```
Comment (JusSn) [inline design/Limits.md]:
```suggestion
`maxSamplersPerShaderStage = 16;` | [Vulkan](https://vulkan.lunarg.com/doc/view/latest/linux/chunked_spec/chap36.html#limits-required) `maxPerStageDescriptorSamplers` | [#409](https://github.com/gpuweb/gpuweb/issues/409)
```

File: design/Limits.md [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+`maxStorageBuffersPerShaderStage = 4;` | [Vulkan](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#limits-minmax) `maxPerStageDescriptorStorageBuffers` | [#409](https://github.com/gpuweb/gpuweb/issues/409)
```
Comment (JusSn) [inline design/Limits.md]:
```suggestion
`maxStorageBuffersPerShaderStage = 4;` | [Vulkan](https://vulkan.lunarg.com/doc/view/latest/linux/chunked_spec/chap36.html#limits-required) `maxPerStageDescriptorStorageBuffers` | [#409](https://github.com/gpuweb/gpuweb/issues/409)
```

File: design/Limits.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+`maxStorageTexturesPerShaderStage = 4;` | [Vulkan](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#limits-minmax) `maxPerStageDescriptorStorageImages` | [#409](https://github.com/gpuweb/gpuweb/issues/409)
```
Comment (JusSn) [inline design/Limits.md]:
```suggestion
`maxStorageTexturesPerShaderStage = 4;` | [Vulkan](https://vulkan.lunarg.com/doc/view/latest/linux/chunked_spec/chap36.html#limits-required) `maxPerStageDescriptorStorageImages` | [#409](https://github.com/gpuweb/gpuweb/issues/409)
```

File: design/Limits.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+`maxUniformBuffersPerShaderStage = 12;` | [Vulkan](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#limits-minmax) `maxPerStageDescriptorUniformBuffers` | [#409](https://github.com/gpuweb/gpuweb/issues/409)
```
Comment (JusSn) [inline design/Limits.md]:
```suggestion
`maxUniformBuffersPerShaderStage = 12;` | [Vulkan](https://vulkan.lunarg.com/doc/view/latest/linux/chunked_spec/chap36.html#limits-required) `maxPerStageDescriptorUniformBuffers` | [#409](https://github.com/gpuweb/gpuweb/issues/409)
```

**Pull request merged**